src/main/scala/nl/biopet/bioconda/BiocondaRecipe.scala

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2018 Sequencing Analysis Support Core - Leiden University Medical Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package nl.biopet.bioconda
+
+import org.yaml.snakeyaml.Yaml
+import java.io.{File, PrintWriter}
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+
+class BiocondaRecipe(name: String,
+                     version: String,
+                     command: String,
+                     sourceUrl: String,
+                     sourceSha256: String,
+                     runRequirements: Seq[String],
+                     buildRequirements: Seq[String],
+                     testCommands: Seq[String],
+                     homeUrl: String,
+                     license: String,
+                     summary: String,
+                     defaultJavaOptions: Seq[String],
+                     buildNumber: Int = 0,
+                     notes: Option[String] = None) {
+
+  def fileName: String = sourceUrl.split("/").last
+  def wrapperFilename: String = s"$name.py"
+
+  def stringToFile(string: String, file: File): Unit = {
+    val writer = new PrintWriter(file)
+    writer.write(string)
+    writer.close()
+  }
+
+  def createRecipeFiles(dir: File): Unit = {
+    dir.mkdirs()
+    val buildSh = new File(dir, "build.sh")
+    val meta = new File(dir, "meta.yaml")
+    val wrapper = new File(dir, wrapperFilename)
+    stringToFile(metaYaml, meta)
+    stringToFile(buildScript, buildSh)
+    stringToFile(wrapperScript, wrapper)
+    // Wrapper should be executable
+    wrapper.setExecutable(true)
+  }
+
+  def createRecipe(dir: File): Unit = {
+    dir.mkdirs()
+    createRecipeFiles(dir)
+    //Also create a subdirectory with the version
+    createRecipeFiles(new File(dir, version))
+  }
+
+  def metaYaml: String = {
+    val yaml = new Yaml()
+    val meta: java.util.Map[String, Object] = {
```

Comment:
no need for java.util.Map here, Map[String, Any] would be better

Reply:
Yes, but the `dumpAsMap` method from snakeyaml works much much better on java maps.